alcs-frontend/src/app/features/board/dialogs/covenant/create/create-covenant-dialog.component.ts

```diff
@@ -0,0 +1,85 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { ApplicationRegionDto } from '../../../../../services/application/application-code.dto';
+import { ApplicationLocalGovernmentDto } from '../../../../../services/application/application-local-government/application-local-government.dto';
+import { ApplicationLocalGovernmentService } from '../../../../../services/application/application-local-government/application-local-government.service';
+import { ApplicationService } from '../../../../../services/application/application.service';
+import { CardService } from '../../../../../services/card/card.service';
+import { CreateCovenantDto } from '../../../../../services/covenant/covenant.dto';
+import { CovenantService } from '../../../../../services/covenant/covenant.service';
+import { ToastService } from '../../../../../services/toast/toast.service';
+
+@Component({
+  selector: 'app-create-covenant-dialog',
+  templateUrl: './create-covenant-dialog.component.html',
+  styleUrls: ['./create-covenant-dialog.component.scss'],
+})
+export class CreateCovenantDialogComponent implements OnInit {
+  regions: ApplicationRegionDto[] = [];
+  localGovernments: ApplicationLocalGovernmentDto[] = [];
+  isLoading: boolean = false;
+  currentBoardCode: string = '';
+
+  fileNumberControl = new FormControl<string | any>('', [Validators.required]);
+  regionControl = new FormControl<string | null>(null, [Validators.required]);
+  localGovernmentControl = new FormControl<string | null>(null, [Validators.required]);
+  applicantControl = new FormControl<string | any>('', [Validators.required]);
+
+  createForm = new FormGroup({
+    fileNumber: this.fileNumberControl,
+    region: this.regionControl,
+    localGovernment: this.localGovernmentControl,
+    applicant: this.applicantControl,
+  });
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    private dialogRef: MatDialogRef<CreateCovenantDialogComponent>,
+    private covenantService: CovenantService,
+    private cardService: CardService,
+    private applicationService: ApplicationService,
+    private localGovernmentService: ApplicationLocalGovernmentService,
+    private toastService: ToastService
+  ) {}
+
+  ngOnInit(): void {
+    this.currentBoardCode = this.data.currentBoardCode;
+    this.cardService.fetchCodes();
+
+    this.localGovernmentService.list().then((res) => {
+      this.localGovernments = res;
+    });
+
+    this.applicationService.$applicationRegions.subscribe((regions) => {
```

Comment:
I wonder if this will live forever and needs to be destroyed?

Reply:
I'll add onDestroy

Reply:
Missing .pipe(takeUntil(this.$destroy)) before the .subscribe